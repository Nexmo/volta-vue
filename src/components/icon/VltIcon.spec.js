import { shallowMount } from '@vue/test-utils';
import { VltIcon } from '..';

describe('vlt-icon', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallowMount(VltIcon, {
      propsData: {
        icon: 'bin',
      },
    });
  });

  test('is a Vue instance', () => {
    expect(wrapper.isVueInstance()).toBeTruthy();
  });

  test('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot();
  });

  test('is red', () => {
    wrapper.setProps({ color: 'red' });
    expect(wrapper.element).toMatchSnapshot();
  });
});
