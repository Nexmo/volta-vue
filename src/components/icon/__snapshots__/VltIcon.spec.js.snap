// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`vlt-icon is red 1`] = `
<svg
  class="Vlt-icon Vlt-red"
>
  <use
    xlink:href="#V_Vlt-icon-bin"
  />
</svg>
`;

exports[`vlt-icon renders correctly 1`] = `
<svg
  class="Vlt-icon"
>
  <use
    xlink:href="#V_Vlt-icon-bin"
  />
</svg>
`;
